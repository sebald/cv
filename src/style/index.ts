export { default as theme, Theme } from './theme';

export * from './mixins';
export * from './styled-components';
export * from './utils';
