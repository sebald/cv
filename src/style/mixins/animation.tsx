import { StyledMixin } from '../theme';
import { ensureDuration } from '../utils';

import { Animation } from './types';

// https://material.io/guidelines/motion/duration-easing.html
export const curve = {
  standard: 'cubic-bezier(0.4, 0.0, 0.2, 1)',
  deceleration: 'cubic-bezier(0.0, 0.0, 0.2, 1)',
  acceleration: 'cubic-bezier(0.4, 0.0, 1, 1)',
  sharp: 'cubic-bezier(0.4, 0.0, 0.6, 1)',
};

export const animation: StyledMixin<Animation> = ({
  theme,
  animation: name,
  duration = 250,
  easing = 'standard',
  delay = 0,
  mode = 'none',
}) => ({
  animation: `${theme.animation[name]} ${ensureDuration(duration)} ${
    curve[easing]
  } ${ensureDuration(delay)} ${mode}`,
});
