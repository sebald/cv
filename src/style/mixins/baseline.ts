import { ThemedInterpolation } from '../types';
import { pxToRem } from '../utils';

export const baseline: ThemedInterpolation = props =>
  pxToRem(props.theme.baseline);
