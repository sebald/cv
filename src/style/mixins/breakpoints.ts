import { Interpolation } from 'styled-components';
import { css } from '../styled-components';
import { StyledMixin, Theme, ThemedStyledProps } from '../theme';
import { px } from '../utils';

export const breakpoints: StyledMixin = props => props.theme.breakpoints;

export const media = <P = {}>(name: keyof Theme['breakpoints']) => (
  strings: TemplateStringsArray,
  ...interpolations: Interpolation<ThemedStyledProps<P>>[]
) => css`
  @media (min-width: ${props => px(breakpoints(props)[name])}) {
    ${css(strings, ...interpolations)};
  }
`;
