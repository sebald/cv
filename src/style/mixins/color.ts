import { Color, TIF } from '../theme';

export const color = (value: Color): TIF => props => props.theme.palette;
