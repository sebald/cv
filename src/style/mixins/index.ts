export * from './animation';
export * from './breakpoints';
export * from './color';
export * from './space';
export * from './typography';
export * from './width';

export * from './types';
