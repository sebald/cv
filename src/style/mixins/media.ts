import { Interpolation, ThemedStyledProps } from 'styled-components';
import { css } from '../styled-components';
import { Theme, ThemedInterpolation } from '../types';
import { px } from '../utils';

export const breakpoints: ThemedInterpolation = props =>
  props.theme.breakpoints;

export const media = <P = {}>(name: keyof Theme['breakpoints']) => (
  strings: TemplateStringsArray,
  ...interpolations: Interpolation<ThemedStyledProps<P, Theme>>[]
) => css`
  @media (min-width: ${props => px(breakpoints(props)[name])}) {
    ${css(strings, ...interpolations)};
  }
`;
