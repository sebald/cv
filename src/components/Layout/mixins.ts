import { css, media, padding } from '../../style';

export const aside = css`
  width: 100%;
  background: ${props => props.theme.palette.grey['50']};
  ${padding(1)} ${media('small')`
    order: 1;
    width: ${props => `${props.theme.layout.sidebar}px`};
  `};
`;
