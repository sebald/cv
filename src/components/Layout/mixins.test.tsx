import { shallow } from 'enzyme';
import * as React from 'react';

import { styled, theme } from '../../style';
import { aside } from './mixins';

test('render correctly', () => {
  const Component = styled('div')`
    ${aside};
  `;
  const wrapper = shallow(<Component theme={theme} />);
  expect(wrapper).toMatchSnapshot();
});
