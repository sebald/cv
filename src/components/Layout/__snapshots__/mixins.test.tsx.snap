// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
.c0 {
  width: 100%;
  background: #fafafa;
  padding: 1rem;
}

@media (min-width:768px) {
  .c0 {
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
    width: 300px;
  }
}

<div
  className="c0"
/>
`;
